File: 6.3_Text and Image Integration - ChatGPT Picture Description.py
# oding = utf-8
# -*- coding:utf-8 -*-
import openai
# 设置ChatGPT的凭据
openai.api_key = 'YOUR_API_KEY'
# 定义要描述的图像
def build_prompt(image_path):
    prompt = f'Image: {image_path}\nDescribe the image:'
    return prompt
# 从用户输入中提取图像路径
def extract_image_path(user_input):
    image_path = ...
    return image_path
# 让ChatGPT生成描述
def generate_description(image_path):
    prompt = build_prompt(image_path)
    # 调用ChatGPT生成描述
    response = openai.Completion.create(
        engine='text-davinci-003',
        prompt=prompt,
        max_tokens=100,
        n=1,
        stop=None,
        temperature=0.7
    )
# 从ChatGPT响应中提取生成的描述
    description = response.choices[0].text.strip()
    return description
# 读取用户输入并调用ChatGPT生成描述
while True:
    user_input = input('Enter image path (or "quit" to exit): ')
    if user_input.lower() == 'quit':
        break
    description = generate_description(image_path)
    print(description)

File: 6.3_Text and Image Integration - ChatGPT Picture Description.cpp
#include <stdio.h>
#define API_KEY "YOUR_API_KEY"
// 构建请求的URL
void build_url(char* url, const char* image_path) {
    char endpoint[] = "https://api.deepai.org/api/text-generator";
    sprintf(url, "%s?image=%s", endpoint, image_path);
}
// 处理API响应
void handle_response(char* response) {
    // 提取生成的文本描述
    const char* keyword = "\"output\":";
    char* description = strstr(response, keyword);
    if (description) {
        description += strlen(keyword);
        description[strcspn(description, "\"")] = '\0';  // 移除描述中的引号
        printf("Description: %s\n", description);
    }
    else {
        printf("Failed to extract description from API response.\n");
    }
}
// 发送API请求并获取响应
size_t curl_write_callback(char* data, size_t size, size_t nmemb, char* response) {
    size_t response_length = size * nmemb;
    memcpy(response, data, response_length);
    return response_length;
}
int main() {
    char image_path[100];
    char url[200];
    char response[100000];
    printf("Enter image path (or 'quit' to exit): ");
    fgets(image_path, sizeof(image_path), stdin);
    curl_post_request(url, response);
    handle_response(response);
}